bson/src/main/org/bson/codecs/ContainerCodecHelper.java

```diff
@@ -62,6 +66,23 @@ static Object readValue(final BsonReader reader, final DecoderContext decoderCon
         }
     }
 
+    static Codec<?> getCodec(final CodecRegistry codecRegistry, final Type type) {
```

Comment:
This is the guts of the implementation: a recursive algorithm for creating a parameterized codec of any depth.

Reply:
Seems familiar to the Pojo codec parameterizedType handling but cleaner.